src/elements/hx-tooltip/index.spec.js

```diff
@@ -0,0 +1,225 @@
+                const elSelector = 'div#hxTooltip';
+                const id = 'hxTooltip';
+                const component = /** @type {HXTooltipElement} */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+        });
+    });
+
+    describe('verify onCreate method', () => {
+        it('test for default position', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const defaultPosition = 'top-center';
+            const attr = component.hasAttribute('position');
+            const bottomStart = component.getAttribute('position');
+
+            expect(attr).to.be.true;
+            expect(bottomStart).to.equal(defaultPosition);
+        });
+
+        it('test for position offset', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const positionOffset = 20;
+            const offsetValue = component.POSITION_OFFSET;
+
+            expect(offsetValue).to.equal(positionOffset);
+        });
+
+        it('test for checking all positions', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const position = component.getAttribute('position');
+            const changePosition = 'right-middle';
+            component.setShadowPosition(changePosition);
+            const newPosition = component.shadowRoot.querySelector('div').getAttribute('position')
+
+            expect(newPosition).to.not.equal(position);
+            expect(newPosition).to.equal(changePosition);
+        });
+    });
+
+    describe('test $onConnect method', () => {
+        it('should have a randomly generated or assigned ID on render', async () => {
+            const fragment = /** @type {HXTooltipElement} */ await fixture(mockup);
+            const attr = fragment.hasAttribute('id');
+            const tooltipId = fragment.getAttribute('id');
+
+            expect(attr).to.be.true;
+            expect(tooltipId).to.not.be.null;
+        });
+
+        it('should have role attribute default to tooltip', async () => {
+            const roleAttr = 'tooltip';
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const attr = component.hasAttribute('role');
+            const role = component.getAttribute('role');
+
+            expect(attr).to.be.true;
+            expect(role).to.equal(roleAttr);
+        });
+    });
+
+    describe('test <hx-tooltip> getters and setters', () => {
+        it('should be able to set for', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const attr = component.for;
+
+            expect(attr).to.not.be.null;
+        });
```

Comment:
Please update to test:

https://github.com/HelixDesignSystem/helix-ui/blob/6ac257bf730b79eb7b55da76800a7e39def34f85/src/elements/hx-tooltip/index.js#L88-L112

* get controlElement
* get htmlFor
* set htmlFor
* setShadowPosition

Reply:
Changes have been done, Please review